comp/core/secrets/secretsimpl/reload_audit.go

```diff
@@ -0,0 +1,124 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package secretsimpl is the implementation for the secrets component
+package secretsimpl
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"regexp"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/util/filesystem"
+	"github.com/DataDog/datadog-agent/pkg/util/scrubber"
+)
+
+var (
+	cutoffLimit       = time.Hour * 24 * 90 // prune entries older than 90 days
+	timeNowFunc       = time.Now
+	apiKeyStringRegex = regexp.MustCompile(`^[[:xdigit:]]{32}(?:[[:xdigit]]{8})?$`)
+)
+
+// auditRow represents when a single secret is refreshed at a moment in time
+type auditRow struct {
+	When   time.Time `json:"when"`
+	Handle string    `json:"handle"`
+	Value  string    `json:"value,omitempty"`
+}
+
+// addToAuditFile adds rows to the audit file based upon newly refreshed secrets
+func addToAuditFile(filename string, secretResponse map[string]string, origin handleToContext, maxSize int) error {
+	if filename == "" {
+		return nil
+	}
+	now := timeNowFunc().UTC()
+	auditRows, err := loadAuditFile(filename, maxSize)
+	if err != nil {
+		if sizeErr, ok := err.(*filesystem.FileSizeLimitError); ok {
+			return fmt.Errorf(`%s cannot be loaded, file size is too large: %v > %v
+You can work-around this limit by setting the flag "secret_audit_file_max_size" to a larger value`, filename, sizeErr.Size, sizeErr.Limit)
+		}
+		if _, ok := err.(*os.PathError); !ok {
+			return fmt.Errorf(`%s cannot be loaded: %s`, filename, err)
+		}
```

Comment:
Should this be done by `loadAuditFile` ?

Reply:
I'm not sure what you're asking. This code path is treating "file not found" as an okay result, but any other error returned by `loadAuditFile` gets wrapped with the "[filename] cannot be loaded". For example, if the audit file contains invalid json. This seems like the right place for that logic.